### ISSUE TITLE
Can't build Hydra on nixos-unstable

### ISSUE THREAD

**Issue by ElvishJerricco** [NixOS/nixpkgs#35398]
## Issue description

Hydra fails to build on `nixos-unstable` and `master` due to a broken `aws-sdk-cpp` dependency, which appears to be different from the `aws-sdk-cpp` at top level since that builds successfully.

### Steps to reproduce

```bash
$ git clone https://github.com/NixOS/nixpkgs
$ cd nixpkgs
$ nix-build -A hydra
...
builder for ‘/nix/store/jx21mw1cw771h9954jaicwxc7a8ps6mc-aws-sdk-cpp-1.3.22.drv’ failed with exit code 2
cannot build derivation ‘/nix/store/32rxs11yf8bii1yqaysvid7x3xlm08ws-hydra-2017-11-21.drv’: 1 dependencies couldn't be built
error: build of ‘/nix/store/32rxs11yf8bii1yqaysvid7x3xlm08ws-hydra-2017-11-21.drv’ failed
```

## Technical details

```
$ nix-store --read-log /nix/store/jx21mw1cw771h9954jaicwxc7a8ps6mc-aws-sdk-cpp-1.3.22.drv
unpacking sources
unpacking source archive /nix/store/8446r3j8mw7xzmam785r0cvdxxd48r8q-source
source root is source
patching sources
configuring
fixing cmake files...
cmake flags: -DCMAKE_BUILD_TYPE=Release -DCMAKE_SKIP_BUILD_RPATH=ON -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/6navrsgn0qnsjgk4vck73phkj7nxn32v-aws-sdk-cpp-1.3.22-dev/include -DCMAKE_INSTALL_LIBDIR=/nix/store/lrly4rrh3kqnql51qph762bw54b51swg-aws-sdk-cpp-1.3.22/lib -DCMAKE_INSTALL_NAME_DIR=/nix/store/lrly4rrh3kqnql51qph762bw54b51swg-aws-sdk-cpp-1.3.22/lib -DCMAKE_INSTALL_PREFIX=/nix/store/lrly4rrh3kqnql51qph762bw54b51swg-aws-sdk-cpp-1.3.22 -DCUSTOM_MEMORY_MANAGEMENT=0 -DBUILD_ONLY=s3 
-- TARGET_ARCH not specified; inferring host OS to be platform compilation target
-- Generating linux build config
-- Dynamic linking enabled
-- Http client: Curl
-- Encryption: Openssl
-- The C compiler identification is GNU 7.3.0
-- The CXX compiler identification is GNU 7.3.0
-- Check for working C compiler: /nix/store/gqg2vrcq7krqi9rrl6pphvsg81sb8pjw-gcc-wrapper-7.3.0/bin/gcc
-- Check for working C compiler: /nix/store/gqg2vrcq7krqi9rrl6pphvsg81sb8pjw-gcc-wrapper-7.3.0/bin/gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /nix/store/gqg2vrcq7krqi9rrl6pphvsg81sb8pjw-gcc-wrapper-7.3.0/bin/g++
-- Check for working CXX compiler: /nix/store/gqg2vrcq7krqi9rrl6pphvsg81sb8pjw-gcc-wrapper-7.3.0/bin/g++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Install binary prefix: linux/intel64
-- Could NOT find Git (missing: GIT_EXECUTABLE) 
-- Custom memory management disabled
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_INSTALL_INCLUDEDIR
    CMAKE_INSTALL_LIBDIR


-- Build files have been written to: /tmp/nix-build-aws-sdk-cpp-1.3.22.drv-0/source/build
building
build flags: -j4 -l4 SHELL=/nix/store/zqh3l3lyw32q1ayb15bnvg9f24j5v2p0-bash-4.4-p12/bin/bash
Scanning dependencies of target aws-cpp-sdk-core
[  1%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/AmazonStreamingWebServiceRequest.cpp.o
[  2%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/AmazonSerializableWebServiceRequest.cpp.o
[  4%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/AmazonWebServiceRequest.cpp.o
[  5%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/Region.cpp.o
[  7%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/Version.cpp.o
[  8%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/external/tinyxml2/tinyxml2.cpp.o
[ 10%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/external/json-cpp/jsoncpp.cpp.o
[ 11%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/auth/AWSAuthSigner.cpp.o
[ 13%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/auth/AWSCredentialsProvider.cpp.o
[ 14%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/auth/AWSCredentialsProviderChain.cpp.o
[ 16%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/client/AWSClient.cpp.o
[ 17%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/client/AWSErrorMarshaller.cpp.o
[ 19%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/client/ClientConfiguration.cpp.o
[ 20%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/client/CoreErrors.cpp.o
[ 22%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/client/DefaultRetryStrategy.cpp.o
[ 23%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/internal/EC2MetadataClient.cpp.o
[ 25%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/http/HttpClient.cpp.o
[ 26%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/http/HttpClientFactory.cpp.o
[ 28%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/http/HttpRequest.cpp.o
[ 29%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/http/HttpTypes.cpp.o
[ 31%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/http/Scheme.cpp.o
[ 32%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/http/URI.cpp.o
[ 34%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/http/standard/StandardHttpRequest.cpp.o
[ 35%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/http/standard/StandardHttpResponse.cpp.o
[ 37%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/utils/DateTime.cpp.o
[ 38%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/utils/HashingUtils.cpp.o
[ 40%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/utils/OSVersionInfo.cpp.o
[ 41%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/utils/StringUtils.cpp.o
[ 43%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/utils/base64/Base64.cpp.o
[ 44%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/utils/crypto/Factories.cpp.o
[ 46%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/utils/crypto/MD5.cpp.o
[ 47%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/utils/crypto/Sha256.cpp.o
[ 49%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/utils/crypto/Sha256HMAC.cpp.o
[ 50%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/utils/json/JsonSerializer.cpp.o
[ 52%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/utils/threading/Executor.cpp.o
[ 53%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/utils/xml/XmlSerializer.cpp.o
[ 55%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/utils/stream/ResponseStream.cpp.o
[ 56%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/utils/logging/AWSLogging.cpp.o
[ 58%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/utils/logging/ConsoleLogSystem.cpp.o
[ 59%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/utils/logging/DefaultLogSystem.cpp.o
[ 61%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/utils/logging/FormattedLogSystem.cpp.o
[ 62%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/utils/logging/LogLevel.cpp.o
[ 64%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/utils/memory/AWSMemory.cpp.o
[ 65%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/http/curl/CurlHandleContainer.cpp.o
[ 67%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/http/curl/CurlHttpClient.cpp.o
[ 68%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/utils/crypto/openssl/CryptoImpl.cpp.o
/tmp/nix-build-aws-sdk-cpp-1.3.22.drv-0/source/aws-cpp-sdk-core/source/http/curl/CurlHttpClient.cpp: In static member function 'static size_t Aws::Http::CurlHttpClient::ReadBody(char*, size_t, size_t, void*)':
/tmp/nix-build-aws-sdk-cpp-1.3.22.drv-0/source/aws-cpp-sdk-core/source/http/curl/CurlHttpClient.cpp:347:37: error: '*' in boolean context, suggest '&&' instead [-Werror=int-in-bool-context]
     if (ioStream != nullptr && size * nmemb)
                                ~~~~~^~~~~~~
[ 70%] Building CXX object aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/utils/unix/FileSystemUtils.cpp.o
At global scope:
cc1plus: error: unrecognized command line option '-Wno-unused-private-field' [-Werror]
cc1plus: error: unrecognized command line option '-Wno-unused-private-field' [-Werror]
cc1plus: all warnings being treated as errors
make[2]: *** [aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/build.make:1119: aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/source/http/curl/CurlHttpClient.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:86: aws-cpp-sdk-core/CMakeFiles/aws-cpp-sdk-core.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
```

 - system: `"x86_64-linux"`
 - host os: `Linux 4.15.0, NixOS, 17.09.git.c831224528c (Hummingbird)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 1.11.16`
 - channels(root): `""`
 - channels(will): `""`
 - nixpkgs: `/nix/store/iwg0swifgivc855hxw06vzl5zbsd819a-nixpkgs-channels`

**Comment by edolstra** [comment]
This is fixed by #35125.

**Comment by vcunat** [comment]
There might be more hydra instances than anticipated, given the upsurge of duplicate reports :-)
